### PR TITLE
Release navigation-1775

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR releases the new navigation (`navigation-1775`) for everyone and removes all the legacy code from the previous navigation. In addition this PR,
- Fixes some minor top navigation issues to support natural command + clicks.
- Changes default password for tests to be compliant with the minimum 8-char length.
- Removes some unused old code.
- Removes `event.stopPropagation` from `Link` component because what we actually want is just to stop the default behavior, not prevent the event to be propagated to upper components. This mainly affected links in dropdowns, tooltips, popovers, etc.

## Checklist

- [X] N/A. All querysets/queries filter by Organization, by Team, and by User
- [X] N/A. Django backend tests
- [X] N/A. Jest frontend tests
- [X] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/lib/components/Link.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { router } from 'kea-router'
 export interface LinkProps extends HTMLProps<HTMLAnchorElement> {
     to: string
     preventClick?: boolean
-    tag?: string | React.Component
+    tag?: string | React.ReactNode
```
Comment (paolodamico) [inline frontend/src/lib/components/Link.tsx:7]: Help needed here

File: posthog/management/commands/setup_dev.py [hunk 2]
```diff
@@ -16,7 +13,7 @@ def handle(self, *args, **options):
             organization, team, user = User.objects.bootstrap(
                 company_name=ORGANIZATION_NAME,
                 email="test@posthog.com",
-                password="pass",
+                password="12345678",
```
Comment (macobo) [inline posthog/management/commands/setup_dev.py:16]: Why? Seems unrelated to this PR.

File: frontend/src/lib/components/Link.tsx [hunk 3]
```diff
@@ -16,15 +16,14 @@ export function Link({ to, preventClick = false, tag = 'a', ...props }: LinkProp
 
         if (!props.target) {
             event.preventDefault()
-            event.stopPropagation()
```
Comment (macobo) [inline frontend/src/lib/components/Link.tsx:19]: Why?

File: frontend/src/lib/components/Link.tsx [hunk 4]
```diff
@@ -16,15 +16,14 @@ export function Link({ to, preventClick = false, tag = 'a', ...props }: LinkProp
...
             if (to && to !== '#' && !preventClick) {
                 router.actions.push(to) // router is mounted automatically, so this is safe to call
             }
         }
         props.onClick && props.onClick(event)
     }
 
-    return React.createElement(tag, {
+    return React.createElement(tag as string, {
```
Comment (macobo) [inline frontend/src/lib/components/Link.tsx:26]: This is lying to the compiler, not sure if this works at all :D 

The proper way here seems to be to be use React.cloneElement for non-strings.

Q: Should this be part of this PR?

File: cypress/integration/licenses.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 describe('Licenses', () => {
     it('Licenses loaded', () => {
-        cy.get('[data-attr=menu-item-settings]').click()
-        cy.get('[data-attr=menu-item-instance-licenses]').click()
+        cy.get('.whoami').click() // Top navigation dropdown
```
Comment (macobo) [inline cypress/integration/licenses.js:3]: Note: Using class names in integration tests is mixing concerns. E.g. here you need to add a comment for the code to be understandable.

Why not add a data-attr?

File: frontend/src/lib/components/Link.tsx (hunk 4 above)
Comment (paolodamico) [reply frontend/src/lib/components/Link.tsx:26]: Tagging @yakkomajuri here as he helped with this change. It is definitely unrelated to the PR, but because I made a small change to this file, tests won't pass unless this TS issue is fixed.

File: posthog/management/commands/setup_dev.py (hunk 2 above)
Comment (paolodamico) [reply posthog/management/commands/setup_dev.py:16]: Agree that is tangential to the PR, but it's a minor change only intended to aid when running tests locally. This makes sure you can create the test user using the UI too and the password to match.

File: frontend/src/lib/components/Link.tsx (hunk 3 above)
Comment (paolodamico) [reply frontend/src/lib/components/Link.tsx:19]: The context on this change is that when we have links inside tooltips, popovers, dropdowns (referring to Ant components), the elements do not get dismissed after clicking the link, because the click does not get propagated. I do believe it makes sense to propagate the event normally. The reason for the `Link` component is to use `kea-router` instead of a browser's native navigation, this is achieved with only the `event.preventDefault()`.

In addition, this component creates an `a` tag for most cases and even when doing a `LinkButton` (which uses an underlying `button` component) forms won't get submitted.

The above being said, there might be a case I might be failing to consider.

File: posthog/management/commands/setup_dev.py (hunk 2 above)
Comment (macobo) [reply posthog/management/commands/setup_dev.py:21]: Ack makes sense. This can cause some slight pain (e.g. heroku deploy password has changed) but that makes sense.

File: frontend/src/lib/components/Link.tsx (hunk 3 above)
Comment (macobo) [reply frontend/src/lib/components/Link.tsx:19]: Hmm. I would have expected LinkButton to be breaking - if not I don't see this breaking anything, though it is a change which might sneakily break something.

**Review by macobo — APPROVED** [review]

File: frontend/src/lib/components/Link.tsx (hunk 3 above)
Comment (paolodamico) [reply frontend/src/lib/components/Link.tsx:19]: I specifically tested `LinkButton` and it did not trigger a form submission, but I do agree we might be inadvertently introducing a bug somewhere. Though I think it's a good opportunity to make this change and fix later if something actually breaks. The alternative is just defaulting to stopPropagation and adding an extra prop that can stop that behavior (I actually did this at first), but seems like it would introduce painful tech debt.

File: frontend/src/lib/components/Link.tsx (hunk 4 above)
Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: Yes it is indeed lying to the compiler - But we were passing a type as a `tag` before that wasn't compatible with `createElement` at all. 

Let me take a look at using `cloneElement`

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: `cloneElement` also doesn't accept the types we're using for the button in `LinkButton`: 

`ReactNode` as what we currently have and `CompoundedComponent` the actual type provided by antd

Comment (paolodamico) [reply frontend/src/lib/components/Link.tsx:26]: Thanks for looking into this @yakkomajuri, how about we keep this workaround for now and add a note to fix this later?

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: If @macobo has time I'd love to learn from him the best approach here. I know you're eager to merge (rightfully so) but I'd like to see if he has a quick fix. Will ping on Slack

Comment (macobo) [reply frontend/src/lib/components/Link.tsx:26]: I think the `correct` way here (if we wanted to support both string and React.Node) is something like the following:

```ts
const props = { ... }
if (typeof tag === 'string') {
  return React.createElement(tag, props)
} else {
  return React.cloneElement(tag, props)
}
```

Not 100% sure this would work though, would need to test it

Comment (paolodamico) [reply frontend/src/lib/components/Link.tsx:26]: Happy to wait to fix this the right way 😁

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: `cloneElement` doesn't work on the tag being a `CompoundedComponent` or `React.ReactNode`

Comment (macobo) [reply frontend/src/lib/components/Link.tsx:26]: Or alternatively, if tag is a class:

```
return <tag {...props} />
```

Might work

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: Ah so gotta change the callsite

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx:26]: let me test

Comment (yakkomajuri) [reply frontend/src/lib/components/Link.tsx]: Ok so that worked but had to change the button's type to `React.FunctionComponentElement<any>` - any objections @macobo ?

**Review by yakkomajuri — APPROVED** [review]
